_posts/2021-01-30-Writing-Better-Code-Cloud-Code-with-Parse-Server-Example.md

```diff
@@ -0,0 +1,191 @@
+
+# Testing
+
+As your Parse Server project gets larger and more complex, it can be challenging to add new features without previous ones becoming affected.
+
+Automated testing is a good way to be assured that your functions are running as expected.
+
+On the latest [Parse Server Example](https://github.com/parse-community/parse-server-example) project, you can create tests by creating `*.spec.js` files in `/spec`. 
+
+This will boot up a mock database, start a Parse Server based off your `index.js`. You don't have to worry about tests impacting your production database.
+
+We have written some example tests to show you how it works.
+
+Now, run `npm run test` and `jasmine` (the testing framework) will check to make sure the tests run as expected. 
+
+Some helpful tips:
+- If you create a test using `fit()` or `fdescribe()`, jasmine will focus on those tests only, and others will be ignored.
+- You can specify a file to test by using `npm run test /spec/testFile.spec.js`
+- You won't be able to access any live data through the tests. You'll have to save mock data using `beforeAll()` if you want to test queries.
+
+# Coverage
+
+Now you've learnt testing, it's important to make sure as much of your cloud functions are covered by tests, so you can be sure that they are all running as expected.
+
+After you've written and tested a few tests, you can run `npm run coverage`. This will create a new folder `/coverage`. Navigate to `/coverage/lcov-report` and open `index.html`, and you'll be able to visually see what bits of your code you need to write tests for, and which sections are covered. The more of your cloud code you cover, the more you can be assured that changes don't have any unforseen consequences.
+
+# Watching Cloud Code
+
+Traditionally, you have to restart your Parse Server whenever you make changes to cloud code. Now, you can type `npm run watch`, and your Parse Server will auto-restart when any affected files are changed.
+
+# New Scripts
+
+* `npm run watch` will start your Parse Server and restart if you make any changes.
+* `npm run lint` will check the linting of your cloud code, tests and `index.js`, as defined in `.eslintrc.json`.
+* `npm run lint-fix` will attempt fix the linting of your cloud code, tests and `index.js`.
+* `npm run prettier` will help improve the formatting and layout of your cloud code, tests and `index.js`, as defined in `.prettierrc`.
+* `npm run test` will run any tests that are written in `/spec`.
+* `npm run coverage` will run tests and check coverage. Output is available in the `/coverage` folder.
+
+# Tying this all together
+
+Great, now you can be assured that your Parse Server's cloud code is working as expected, and is clean as it can be.
+
+If you want to go one step further, you can use GitHub actions to check your codes' linting and tests prior to commit.
```

Comment:
Perhaps add a link for GH actions - a marketing page or a getting started page?

and maybe change to `GitHub Actions or another CI platform`